### PR TITLE
Add a recipe for auto_sklearn

### PR DESCRIPTION
**Pull request by iancoffey**
For folks wanting to experiment with auto_sklearn, its easy to get it cleanly installed with this Dockerfile.

Thanks!

### CODE REVIEW

File: docs/using/recipes.md [hunk 1]
```diff
@@ -516,3 +516,24 @@ RUN pip install jupyter_contrib_nbextensions && \
...
+```
+ARG BASE_CONTAINER=jupyter/scipy-notebook
+FROM jupyter/scipy-notebook:latest
+
+USER root
+
+# autosklearn requires swig, which no other image has
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends swig && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (romainx) [inline docs/using/recipes.md:533]: `apt-get clean && rm -rf /var/lib/apt/lists/*`

File: docs/using/recipes.md [hunk 2]
```diff
@@ -516,3 +516,24 @@ RUN pip install jupyter_contrib_nbextensions && \
...
+
+USER $NB_UID
+ARG py_ver=3.6
```
Comment (romainx) [inline docs/using/recipes.md:536]: Why `ARG py_ver=3.6`? I do not see it used anywhere (note the current python version on Jupyter Docker stack is `3.7.6`).

File: docs/using/recipes.md [hunk 3]
```diff
@@ -516,3 +516,24 @@ RUN pip install jupyter_contrib_nbextensions && \
...
+
+RUN pip install auto-sklearn
```
Comment (romainx) [inline docs/using/recipes.md:538]: `pip install --quiet`

File: docs/using/recipes.md [hunk 4]
```diff
@@ -516,3 +516,24 @@ RUN pip install jupyter_contrib_nbextensions && \
 
 Ref:
 [https://github.com/jupyter/docker-stacks/issues/675](https://github.com/jupyter/docker-stacks/issues/675)
+
+## Enable auto-sklearn notebooks
+
+Using auto-sklearn requires swig, which the other notebook images lack, so it cant be experimented with. Also, there is no Conda package for auto-sklearn.
```
Comment (romainx) [inline docs/using/recipes.md:522]: Use inline code for `auto-sklearn`, `swig`

File: docs/using/recipes.md (hunk 3 above)
Comment (romainx) [reply docs/using/recipes.md:538]: Maybe the install could break permissions. Could you check it and if needed add

```dockerfile
 fix-permissions $CONDA_DIR && \
 fix-permissions /home/$NB_USER
```

File: docs/using/recipes.md (hunk 2 above)
Comment (iancoffey) [reply docs/using/recipes.md:536]: Good catch, thanks!

File: docs/using/recipes.md (hunk 3 above)
Comment (iancoffey) [reply docs/using/recipes.md:538]: I tested this again, the install did not seem to break perms, I added the quiet as well. Thank you for the input.

File: docs/using/recipes.md (hunk 4 above)
Comment (iancoffey) [reply docs/using/recipes.md:522]: Added, thanks!

**Comment by iancoffey** [comment]
@romainx I think this is ready, thank you!

File: docs/using/recipes.md [hunk 5]
```diff
@@ -530,10 +530,11 @@ USER root
...
     apt-get install -y --no-install-recommends swig && \
+    apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
+
 USER $NB_UID
-ARG py_ver=3.6
 
-RUN pip install auto-sklearn
+RUN pip install --quiet auto-sklearn
```
Comment (romainx) [inline docs/using/recipes.md:539]: Sorry but I have to mention a remark that has been done just a couple of days before on another case #1082. It's a good practice to add the `--no-cache-dir` option to reduce the image size. Maybe you could add it?

Comment (iancoffey) [reply docs/using/recipes.md:539]: No problem, added!

Comment (romainx) [reply docs/using/recipes.md:539]: Tks ping me when it's done.

Comment (iancoffey) [reply docs/using/recipes.md]: Done, thanks

**Pull request merged**